src/main/java/core/mate/academy/service/MachineServiceImpl.java

```
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<Bulldozer> bulldozers = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozers);
+        }
+
+        if (type == Track.class) {
+            List<Track> tracks = new TrackProducer().get();
+            return new ArrayList<>(tracks);
+        }
+
+        if (type == Excavator.class) {
+            List<Excavator> excavators = new ExcavatorProducer().get();
+            return new ArrayList<>(excavators);
+        }
+        return null;
```

return `null` isn't a good practice
let's throw an exception or return Collections.emptyList()